apps/src/lab2/views/dialogs/DialogManager.tsx

```diff
@@ -37,17 +48,37 @@ const DialogManager: React.FunctionComponent<DialogManagerProps> = ({
   children,
 }) => {
   const [openDialog, setOpenDialog] = useState<DialogType | null>(null);
+  const [shouldThrowOnCancel, setShouldThrowOnCancel] =
+    useState<boolean>(false);
```

Comment:
if it's handy for you, you can have the dialog reject the promise (potentially throwing an exception if you've await'ed it) when the user presses `cancel`. `confirm`/`neutral` will resolve it.

Defaults to `false` so that all button presses resolve.